### PR TITLE
Use verify prop to gate instead of ICN check

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
We can use the usual va.gov verification logic to make sure a user has the necessary profile info, instead of checking for ICNs in the identifiers service, like legacy VAOS does.

## Testing done
Local testing

## Acceptance criteria
- [ ] People without verified accounts linked to an MVI record are not allowed to use the tool

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/containers/VAOSApp.jsx [hunk 1]
```diff
@@ -56,12 +56,12 @@ export class VAOSApp extends React.Component {
 
     return (
       <RequiredLoginView
-        authRequired={1}
         serviceRequired={[
           backendServices.USER_PROFILE,
           backendServices.FACILITIES,
         ]}
         user={user}
+        verify
```
Comment (jbalboni) [inline src/applications/vaos/containers/VAOSApp.jsx]: This triggers the verification logic in this component, which will check the MVI profile status, which should correlate with the user having an ICN or not.

**Review by kreek — APPROVED** [review]

File: src/applications/vaos/tests/containers/VAOSApp.unit.spec.jsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ describe('VAOS <VAOSApp>', () => {
   it('should render login gate and downtime notification', () => {
```
Comment (narin) [inline src/applications/vaos/tests/containers/VAOSApp.unit.spec.jsx]: Would it be useful to also add a test where `verified = false`?

**Review by narin — APPROVED** [review]
Minor feedback but otherwise LGTM!

File: src/applications/vaos/tests/containers/VAOSApp.unit.spec.jsx (hunk 2 above)
Comment (jbalboni) [reply src/applications/vaos/tests/containers/VAOSApp.unit.spec.jsx]: Probably wouldn't hurt.

**Review by jhonny218 — APPROVED** [review]
LGTM

**Pull request merged**